src/time_integration/paired_explicit_runge_kutta/paired_explicit_runge_kutta.jl

```diff
@@ -5,8 +5,176 @@
 @muladd begin
 #! format: noindent
 
-# Basic implementation of the second-order paired explicit Runge-Kutta (PERK) method
-include("methods_PERK2.jl")
 # Define all of the functions necessary for polynomial optimizations
 include("polynomial_optimizer.jl")
+
+# Abstract base type for both single/standalone and multi-level 
+# PERK (Paired-Explicit Runge-Kutta) time integration schemes
+abstract type AbstractPairedExplicitRK end
+# Abstract base type for single/standalone PERK time integration schemes
+abstract type AbstractPairedExplicitRKSingle <: AbstractPairedExplicitRK end
+
+# This struct is needed to fake https://github.com/SciML/OrdinaryDiffEq.jl/blob/0c2048a502101647ac35faabd80da8a5645beac7/src/integrators/type.jl#L1
+mutable struct PairedExplicitRKOptions{Callback, TStops}
+    callback::Callback # callbacks; used in Trixi
+    adaptive::Bool # whether the algorithm is adaptive
+    dtmax::Float64 # ignored
+    maxiters::Int # maximal number of time steps
+    tstops::TStops # tstops from https://diffeq.sciml.ai/v6.8/basics/common_solver_opts/#Output-Control-1; ignored
+end
+
+function PairedExplicitRKOptions(callback, tspan; maxiters = typemax(Int), kwargs...)
+    tstops_internal = BinaryHeap{eltype(tspan)}(FasterForward())
+    # We add last(tspan) to make sure that the time integration stops at the end time
+    push!(tstops_internal, last(tspan))
+    # We add 2 * last(tspan) because add_tstop!(integrator, t) is only called by DiffEqCallbacks.jl if tstops contains a time that is larger than t
+    # (https://github.com/SciML/DiffEqCallbacks.jl/blob/025dfe99029bd0f30a2e027582744528eb92cd24/src/iterative_and_periodic.jl#L92)
+    push!(tstops_internal, 2 * last(tspan))
+    PairedExplicitRKOptions{typeof(callback), typeof(tstops_internal)}(callback,
+                                                                       false, Inf,
+                                                                       maxiters,
+                                                                       tstops_internal)
+end
+
+abstract type PairedExplicitRK end
```

Comment:
Hm I think we should maybe add the `Integrator` here to avoid confusion, i.e, 
`PairedExplicitRKIntegrator` or maybe even `AbstractPairedExplicitRKIntegrator`.